webextension/background.js

```diff
@@ -1,14 +1,23 @@
-browser.tabs.onCreated.addListener(tab => {
-  if (tab.incognito) {
-    disableAddon(tab.id);
-  }
-});
+browser.runtime.sendMessage({
+  method: "getPreference",
+  pref: "browser.privatebrowsing.autostart"
+}).then(pbAutoStart => {
+
+  // We don't want to disable the addon if we are in auto private-browsing.
+  if (!pbAutoStart) {
+    browser.tabs.onCreated.addListener(tab => {
+      if (tab.incognito) {
+        disableAddon(tab.id);
+      }
+    });
 
-browser.tabs.query({}).then(tabs => {
-  for (let tab of tabs) { // eslint-disable-line prefer-const
-    if (tab.incognito) {
-      disableAddon(tab.id);
-    }
+    browser.tabs.query({}).then(tabs => {
+      for (let tab of tabs) { // eslint-disable-line prefer-const
+        if (tab.incognito) {
+          disableAddon(tab.id);
+        }
```

Comment:
Are these the only 2 places we're doing this `if (tab.incognito) { disableAddon(tab.id); }` logic? If we're doing it in more places, maybe we should refactor into a helper function?